python/naiveautoml/evaluators.py

```diff
@@ -0,0 +1,317 @@
+import warnings
+
+from lccv import lccv
+import numpy as np
+import pandas as pd
+import sklearn
+
+from .commons import\
+    get_scoring_name, build_scorer
+
+class Lccv_validator:
+
+    def __init__(self, instance, train_size=0.8):
+        self.r = -np.inf
+        self.instance = instance
+        self.train_size = train_size
+
+    def __call__(self, pl, X, y, scorings, errors="message"):
+        warnings.filterwarnings('ignore', module='sklearn')
+        warnings.filterwarnings('ignore', module='numpy')
+        try:
+            if not isinstance(scorings, list):
+                scorings = [scorings]
+
+            try:
+                score, _, _, elcm = lccv(
+                    pl,
+                    X,
+                    y,
+                    r=self.r,
+                    base_scoring=scorings[0],
+                    additional_scorings=scorings[1:],
+                    target_anchor=self.train_size
+                )
+                if not np.isnan(score) and score > self.r:
+                    self.r = score
+
+                results_at_highest_anchor = elcm.df[elcm.df["anchor"] == np.max(elcm.df["anchor"])].mean(
+                    numeric_only=True)
+                results = {
+                    s: results_at_highest_anchor[f"score_test_{s}"] if not np.isnan(score) else np.nan for s in scorings
+                }
+                return results, self  # return the object itself, so that it can be overwritten in the pool (necessary because of pynisher)
+            except KeyboardInterrupt:
+                raise
+            except Exception:
+                if errors == "message":
+                    self.instance.logger.info(f"Observed exception in validation of pipeline {pl}.")
+                else:
+                    raise
+            return {
+                s: np.nan for s in scorings
+            }
+        except KeyboardInterrupt:
+            raise
+        except Exception as e:
+            if errors in ["message", "ignore"]:
+                if errors == "message":
+                    self.instance.logger.error(f"Observed an error: {e}")
+                return None
+            else:
+                raise
+
+class Kfold_5:
```

Comment:
Is there any reason why we should have different classes for Kfold and MCCV only because of different fold numbers?

Reply:
Actually there is no reason, I already solved that and refactored the code, now there is a general class for KFold and another for Mccv.